distributed/client.py

```diff
@@ -4007,6 +4007,36 @@ async def _get_task_stream(
         else:
             return msgs
 
+    async def _register_scheduler_plugin(self, plugin=None, **kwargs):
+        if isinstance(plugin, type):
+            plugin = plugin(**kwargs)
+
+        return await self.scheduler.register_scheduler_plugin(
+            plugin=dumps(plugin, protocol=4)
+        )
+
+    def register_scheduler_plugin(self, plugin=None, **kwargs):
```

Comment:
Users should always need to specify `plugin` (whether it's a class or an instance)

```suggestion
    def register_scheduler_plugin(self, plugin, **kwargs):
```